### PR TITLE
remove ThirdParty API from v1

### PR DESCRIPTION
**Pull request by caesarxuchao**
Removing ThirdParty API related types are from v1/types.go

@lavalamp @brendandburns @bgrant0607

### CODE REVIEW

File: pkg/api/v1/register.go [hunk 1]
```diff
@@ -132,6 +129,3 @@ func (*ComponentStatus) IsAnAPIObject()           {}
 func (*ComponentStatusList) IsAnAPIObject()       {}
 func (*SerializedReference) IsAnAPIObject()       {}
 func (*RangeAllocation) IsAnAPIObject()           {}
-func (*ThirdPartyResource) IsAnAPIObject()        {}
```
Comment (lavalamp) [inline pkg/api/v1/register.go]: Remove the actual types, too. They should be declared in experimental. Right?

Comment (caesarxuchao) [inline pkg/api/v1/register.go]: Thanks. I pushed another commit to remove them form api/

**Comment by ncdc** [comment]
@smarterclayton

**Comment by smarterclayton** [comment]
Does this mean they won't be persisted if the server's storage version is v1?

**Comment by lavalamp** [comment]
Experimental should have its own storage version-- @caesarxuchao is fixing that in another PR.

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
Squash please though

**Comment by caesarxuchao** [comment]
Squashed :)

**Comment by lavalamp** [comment]
@k8s-oncall @brendandburns I want this for 1.1

**Comment by lavalamp** [comment]
LGTM

**Comment by caesarxuchao** [comment]
Rebased.

**Comment by lavalamp** [comment]
LGTM again

**Pull request merged**